packit_service/service/api/utils.py

```diff
@@ -0,0 +1,13 @@
+from flask import make_response
+from http import HTTPStatus
+from json import dumps
+
+
+def response_maker(result, content_range=None, status=HTTPStatus.OK):
+    """response_maker is a wrapper around flask's make_response"""
+    resp = make_response(dumps(result), status)
+    if content_range:
```

Comment:
Doing like this it might make future updates cumbersome, as it would force us to update this method whenever we want to add a new optional header.

How about `response_maker` adding only the common headers and let the individual API endpoints to complete that with anything other they need and explicitly set the status code?

Reply:
This returns a response so instead of doing
`return response_maker(params.....)` 
we can add additional headers by doing

```
response = response_maker(params.....)
response.headers["something"] = "blah blah"
return response
```

Reply:
yes, exactly, and this is why I'm saying to remove the `content_range` argument and rather add it when it's needed :)

Reply:
Oh yeah 🤦